realm/realm-library/src/main/java/io/realm/ObjectChangeSet.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+/**
+ * Information about the changes made to an object.
+ * @see RealmObject#addChangeListener(RealmObjectChangeListener) .
+ */
+public interface ObjectChangeSet {
+
+    /**
+     * Information about a specific field which changed in an {@link RealmObject} change notification.
+     */
+    class FieldChange {
+        /**
+         * The name of the field which changed.
+         */
+        final String fieldName;
+
+        public FieldChange(String fieldName) {
+            this.fieldName = fieldName;
+        }
+    }
+
+    /**
+     * @return {@code null} if the object has been deleted. Otherwise returns the information about changed fields.
+     */
+    FieldChange[] getFieldChanges();
```

Comment:
Having `isDeleted()`, `getChangedFields()`, `isChanged(String fieldName)` and `getChange(type, fieldName)` probably covers the use cases I can think of 👍 

It seems a rather big (and arbitrary) limitation that this doesn't work for local transactions. Any reason why? 

IMO we should just document that you need to be careful if using this with objects with _big_  field values. You shouldn't register 100's of these anyway, but doing it on an object with a 16MB image is probably a bad idea (if it changes). Cocoa has a feature request for ignoring certain fields I believe, but this is currently out of scope.